src/util/getNextThoughtsWithContextChain.js

```diff
@@ -0,0 +1,178 @@
+import { store } from '../store.js'
+
+import {
+  RANKED_ROOT,
+  ROOT_TOKEN,
+} from '../constants.js'
+// util
+import {
+  contextOf,
+  getContextsSortedAndRanked,
+  getSortPreference,
+  getThought,
+  getThoughtsRanked,
+  getThoughtsSorted,
+  head,
+  isFunction,
+  meta,
+  pathToContext,
+  splitChain,
+  nextSibling as thoughtNextSibling,
+  unroot,
+} from '../util.js'
+import { isContextViewActive } from './isContextViewActive.js'
+
+const contextToThought = (context, thoughtIndex) => context.map(contextValue => {
+  const thought = getThought(contextValue, thoughtIndex)
+  return { value: thought.value, rank: thought.contexts[0] && thought.contexts[0].rank }
+})
+
+const mapContextsToThoughts = (contexts, thoughtIndex) => contexts.map(({ context }) => contextToThought(context, thoughtIndex))
+
+const isValidContextView = context => {
+  // const allowSingleContext = false
+  if (!context.length) {
+    return false
+  }
+  return isContextViewActive(context) && getContextsSortedAndRanked(head(pathToContext(context))).length > 1
+}
+
+const getContextSiblings = (context, thoughtIndex) => {
+  const contexts = getContextsSortedAndRanked(head(context))
+  return mapContextsToThoughts(contexts, thoughtIndex)
+}
+
+const nextSiblingContext = (value, rank, context, thoughtIndex) => {
+  const contextSiblings = getContextSiblings(context, thoughtIndex)
+  const currentIndex = contextSiblings.findIndex(thoughts => thoughts.find(thought => thought.value === value))
+  return contextSiblings[currentIndex + 1]
+}
+
+const firstChildOfContext = (value, cursor, thoughtIndex) => {
```

Comment:
All of the references to `cursor` should simply be to `path`. It doesn't have to be the `cursor`.